crates/rethnet_rpc_server/src/methods.rs

```diff
@@ -0,0 +1,781 @@
+#[serde(tag = "method", content = "params")]
+#[allow(clippy::large_enum_variant)]
+enum HardhatMethodInvocation {
+    #[serde(rename = "hardhat_addCompilationResult")]
+    AddCompilationResult(
+        /// solc version:
+        String,
+        compiler_io::CompilerInput,
+        compiler_io::CompilerOutput,
+    ),
+    #[serde(
+        rename = "hardhat_dropTransaction",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    DropTransaction(B256),
+    #[serde(rename = "hardhat_getAutomine")]
+    GetAutomine(),
+    #[serde(rename = "hardhat_getStackTraceFailuresCount")]
+    GetStackTraceFailuresCount(),
+    #[serde(
+        rename = "hardhat_impersonateAccount",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    ImpersonateAccount(Address),
+    #[serde(rename = "hardhat_intervalMine")]
+    IntervalMine(),
+    #[serde(rename = "hardhat_metadata")]
+    Metadata(),
+    #[serde(rename = "hardhat_mine")]
+    Mine(
+        /// block count:
+        U256,
+        /// interval:
+        U256,
+    ),
+    #[serde(
+        rename = "hardhat_reset",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    Reset(Option<RpcHardhatNetworkConfig>),
+    #[serde(rename = "hardhat_setBalance")]
+    SetBalance(Address, U256),
+    #[serde(rename = "hardhat_setCode")]
+    SetCode(Address, ZeroXPrefixedBytes),
+    #[serde(
+        rename = "hardhat_setCoinbase",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetCoinbase(Address),
+    #[serde(
+        rename = "hardhat_setLoggingEnabled",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetLoggingEnabled(bool),
+    #[serde(
+        rename = "hardhat_setMinGasPrice",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetMinGasPrice(U256),
+    #[serde(
+        rename = "hardhat_setNextBlockBaseFeePerGas",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetNextBlockBaseFeePerGas(U256),
+    #[serde(rename = "hardhat_setNonce")]
+    SetNonce(Address, U256),
+    #[serde(
+        rename = "hardhat_setPrevRandao",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    SetPrevRandao(ZeroXPrefixedBytes),
+    #[serde(rename = "hardhat_setStorageAt")]
+    SetStorageAt(Address, U256, ZeroXPrefixedBytes),
+    #[serde(
+        rename = "hardhat_stopImpersonatingAccount",
+        serialize_with = "single_to_sequence",
+        deserialize_with = "sequence_to_single"
+    )]
+    StopImpersonatingAccount(Address),
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Deserialize, serde::Serialize)]
+struct RpcHardhatNetworkConfig {
```

Comment:
addressed in https://github.com/NomicFoundation/hardhat/pull/3998/commits/7edf75ec9dc52682479b08a4f8e5d82256a72a71, though i chose to use a different name than `rpc`, since that's such an overloaded term in this crate.